pages/trial.tsx

```diff
@@ -0,0 +1,372 @@
+              <Image
+                src="/images/logo-and-wordmark-black-words.png"
+                alt="Grouparoo Logo"
+                width={150}
+                height={32}
+                className="mb-3"
+              />
+            </Link>
+            <h1>Grouparoo Cloud</h1>
+            <h2>Start Your Free 30 Day Trial</h2>
+            <p>No credit card required.</p>
+          </Col>
+          <Col
+            xs={12}
+            md={9}
+            xl={7}
+            className="d-flex justify-content-around mx-auto m-0 "
+          >
+            <Error message={error.generic} />
+            {!registered ? (
+              <Container className="bg-white border shadow border-3 rounded col-11 p-4 mt-4 mt-md-0">
+                <h3 className="mx-auto text-center">
+                  Create Your Grouparoo Subdomain
+                </h3>
+
+                <Form id="form" onSubmit={handleSubmit(onSubmit)}>
+                  {" "}
+                  <Form.Group>
+                    <Form.Label>First Name</Form.Label>
+                    <Form.Control
+                      {...register("firstName")}
+                      type="text"
+                      name="firstName"
+                      defaultValue=""
+                      disabled={disabled}
+                    />
+                  </Form.Group>
+                  <Form.Group>
+                    <Form.Label>Last Name</Form.Label>
+                    <Form.Control
+                      {...register("lastName")}
+                      type="text"
+                      name="lastName"
+                      defaultValue=""
+                      disabled={disabled}
+                    />
+                  </Form.Group>
+                  <Form.Group>
+                    <Form.Label>Work Email *</Form.Label>
+                    <Form.Control
+                      {...register("email", { required: true })}
+                      type="email"
+                      name="email"
+                      defaultValue=""
+                      disabled={disabled}
+                    />
+                    <Error message={error.email} />
+                    {errors.email && (
+                      <small style={{ color: "red" }}>
+                        A valid email is required.
+                      </small>
+                    )}
+                  </Form.Group>
+                  <Form.Group>
+                    <Form.Label>Company Website *</Form.Label>
+                    <Form.Control
+                      {...register("companyWebsite", {
+                        required: true,
+                        pattern: urlPattern,
+                      })}
+                      type="text"
+                      name="companyWebsite"
+                      defaultValue=""
+                      onChange={onSubdomainChange}
+                      disabled={disabled}
+                    />
+                  </Form.Group>
```

Comment:
Perhaps... we could just get the subdomain from the email address?  If I'm `evan@foo.com` it seems like a likely guess that the subdomain I want is `foo.grouparoo.app`?

Reply:
In general, I think the validation in use here is too strong anyway.  The subdomain didn't appear until I added `www` in front of `www.company.com`.   

![Screen Shot 2021-06-15 at 4 11 29 PM](https://user-images.githubusercontent.com/303226/122134950-76932080-cdf4-11eb-8d1f-c6ca0408686e.png)

Reply:
Yeah it definitely needs a change!  Andy pointed it out too.  Thoughts on using the email instead of the url to create the domain name @andyjih ?  

I imagine our target user isn't out signing up for things with an '@gmail' or '@outlook' email, but that would be my only concern there... though it's not a big one (people can change it, and the serverside validation would catch duplicates)

Reply:
We have had some people reach out for demo requests with @gmail email addresses, and I generally want to filter those out if possible. So I guess the rough options as I see it are:
1. Use the email domain, but add validation to exclude common personal email domains (gmail, yahoo, aol, outlook, etc.)
2. Keep using company website, but loosen the validation to allow URLs that don't have `www` or `http`.

I lean slightly towards 2, but definitely open to either option!

Reply:
I'm leaning towards 2 given what you shared about demo requests